### PR TITLE
 [K4.0] Hide image for guests - Hint does not appear #3254

### PR DESCRIPTION
**Pull request by xillibit**

### CODE REVIEW

**Comment by 810** [comment]
if you have more attachments, you see the text as many as total attachments.

Also update crypsisb3

**Comment by rich20** [comment]
#### Crypsis

- [ ] It shows also on attachments such as zip "This image is hidden for guests."

#### Blue Eagle

- [ ] Does not show anything.

**Comment by xillibit** [comment]
Try now

File: components/com_kunena/site/views/topic/view.html.php [hunk 1]
```diff
@@ -1,4 +1,3 @@
-<?php
```
Comment (810) [inline components/com_kunena/site/views/topic/view.html.php]: you removed this

**Comment by xillibit** [comment]
Fixed

**Comment by rich20** [comment]
#### Crypsis
- [x]  Show Message (Hide Images)
- [ ] if you have inserted more images, then you see more messages (is it ok?)
- [ ] Show Message (Hide Attachments)

#### Blue Eagle

- [ ] Show Message (Hide Images)
- [ ] Show Message (Hide Attachments)
- [ ] Edit Message - Fatal Error

Fatal Error: Call to a member function getNbAttachments() on null in components/com_kunena/views/topic/view.html.php on line 407

**Comment by xillibit** [comment]
> if you have inserted more images, then you see more messages (is it ok?)

i don't understand

**Comment by rich20** [comment]
In this message are 3 images

bildschirmfoto vom 2015-08-01 17 49 54

**Comment by xillibit** [comment]
Try now

**Comment by rich20** [comment]
#### Crypsis
- [x] Show Message (Hide Images)
- [x] Show Message (Hide Images - embed  via url such as [img]address.png[/img])
- [ ] if you have inserted more images, then you see more messages (but I think, is it ok)
- [x] Show Message (Hide Attachments)

#### Blue Eagle

- [x] Show Message (Hide Images)
- [x] Show Message (Hide Images - embed via url such as [img]address.png[/img])
- [x] Show Message (Hide Attachments)
- [x] Edit Message - Fatal Error

**Comment by rich20** [comment]
#### Crypsis

- [ ] At a image in post, the text "This image.." shows twice (for the image and also for the attachment thumb).

#### Blue Esagle
- [ ] At a image in post, the text "This image.." shows twice
- [ ] If you only add images in messages (without attachments), registered users see both. Image and the text.
bildschirmfoto vom 2015-08-02 16 05 52

**Comment by rich20** [comment]
#### Crypsis

Notice: Undefined variable: attach in components/com_kunena/template/crypsis/layouts/message/item/default.php on line 62

#### Blue Eagle

Notice: Undefined variable: attach in components/com_kunena/template/blue_eagle/html/topic/default_message.php on line 36

**Comment by rich20** [comment]
#### Crypsis

Notice: Trying to get property of non-object in components/com_kunena/template/crypsis/layouts/message/item/default.php on line 61

**Comment by rich20** [comment]
- [x] Success

**Pull request merged**